packages/backup-restore-operator/charts/templates/deployment.yaml

```diff
@@ -0,0 +1,53 @@
+apiVersion: apps/v1 
+kind: Deployment
+metadata:
+  name: backup-restore-operator
+  namespace: {{ .Release.Namespace }}
+  labels:
+    resources.cattle.io/operator: backup-restore
+spec:
+  selector:
+    matchLabels:
+      resources.cattle.io/operator: backup-restore
+  template:
+    metadata:
+      labels:
+        resources.cattle.io/operator: backup-restore
+    spec:
+      serviceAccountName: backup-restore-operator-serviceaccount
+      containers:
+      - name: backup-restore-operator
+        image: {{ template "system_default_registry" . }}{{ .Values.image }}:{{ .Values.tag }}
+        imagePullPolicy: Always
+        env:
+        - name: CHART_NAMESPACE
+          value: {{ .Release.Namespace }}
+          {{- if .Values.pvc }}
+        - name: DEFAULT_PVC_BACKUP_STORAGE_LOCATION
+          value: {{ .Release.Name }}-pvc
+          {{- end }}
+          {{- if .Values.s3 }}
+        - name: DEFAULT_S3_BACKUP_STORAGE_LOCATION
+          value: {{ .Release.Name }}-s3
+          {{- end }}
+        {{- if .Values.pvc }}
+        volumeMounts:
+        - mountPath: "/var/lib/backups"
+          name: pv-storage
+      volumes:
+        - name: pv-storage
+          persistentVolumeClaim:
+            claimName: {{ .Values.pvcName }}
+          {{- end}}
+    {{- with .Values.nodeSelector }}
+    nodeSelector:
+    {{- toYaml . | nindent 8 }}
+    {{- end }}
+    {{- with .Values.affinity }}
+    affinity:
+    {{- toYaml . | nindent 8 }}
+    {{- end }}
+    {{- with .Values.tolerations }}
+    tolerations:
+    {{- toYaml . | nindent 8 }}
+    {{- end }}
```

Comment:
Instead of using `with`, I think it would be easier to use `if` and just `toYaml` the entire `.Values.<selector>` like in `rancher-pushprox` here:

https://github.com/rancher/charts/blob/98d70cee35a19afd217633b5283e21a0dd0b8296/packages/rancher-pushprox/charts/templates/pushprox-clients.yaml#L16-L18

Reply:
Isn't this almost similar? 😅 And `with` allows setting the scope so it's easier to use I think

Reply:
This is almost similar, and `with` allows setting scope so it's easier to use I think, I don't think I need to change this